source/slang/slang-check-modifier.cpp

```diff
@@ -362,6 +362,47 @@ namespace Slang
             numThreadsAttr->y = values[1];
             numThreadsAttr->z = values[2];
         }
+        else if (auto waveSizeAttr = as<WaveSizeAttribute>(attr))
+        {
+            SLANG_ASSERT(attr->args.getCount() == 1);
+
+            IntVal* value = nullptr;
+
+            auto arg = attr->args[0];
+            if (arg)
+            {
+                auto intValue = checkLinkTimeConstantIntVal(arg);
+                if (!intValue)
+                {
+                    return false;
+                }
+                if (auto constIntVal = as<ConstantIntVal>(intValue))
+                {
+                    bool isValidWaveSize = false;
+                    const IntegerLiteralValue waveSize = constIntVal->getValue();
+                    for (int validWaveSize : { 4, 8, 16, 32, 64, 128 })
```

Comment:
I believe the warp sizes should include `1` and `2` if Slang is to fully support ARM GPU's ([Mali specifically](https://tvm.apache.org/2018/01/16/opt-mali-gpu)) since some ARM paired GPU's support independent thread launching:  
```suggestion
                    for (int validWaveSize : { 1, 2, 4, 8, 16, 32, 64, 128 })
```

Otherwise everything looks good.

Reply:
This change is mainly for supporting what SM6.6 spec supports.
The spec document explicitly says the following,
```
The allowed wave sizes that an HLSL shader may specify are the powers of 2 between 4 and 128, inclusive. In other words, the set: [4, 8, 16, 32, 64, 128].
```
We can easily add 1, and 2 if we have a use case later.
I am not going to add anything more than the spec says.